### PR TITLE
Adds probe support to new k8s deployment agents

### PR DESCRIPTION
**Pull request by tlm**
### Checklist

 - [x] Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?
 - [x] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?
 - [x] Do comments answer the question of why design decisions were made?

----

## Description of change

Adds support kubernetes agent probe support

## QA steps

* Run unit tests.

### CODE REVIEW

File: caas/kubernetes/provider/application/application.go [hunk 1]
```diff
@@ -35,12 +35,17 @@ import (
 	"github.com/juju/juju/core/paths"
 	"github.com/juju/juju/core/watcher"
 	jujustorage "github.com/juju/juju/storage"
+	//"github.com/juju/juju/worker/caasprober"
```
Comment (hpidcock) [inline caas/kubernetes/provider/application/application.go:38]: ?

File: caas/kubernetes/provider/constants/constants.go [hunk 2]
```diff
@@ -10,6 +10,13 @@ import (
 )
 
 const (
+	AgentHTTPProbePort = "3856"
```
Comment (hpidcock) [inline caas/kubernetes/provider/constants/constants.go:13]: Comments would be good here.

File: worker/caasprober/prober_test.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caasprober_test
+
+//import (
```
Comment (hpidcock) [inline worker/caasprober/prober_test.go:6]: ?

**Review by hpidcock — APPROVED** [review]
LGTM thank-you for this, just a few little things

**Comment by tlm** [comment]
$$merge$$

**Comment by tlm** [comment]
$$merge$$

**Pull request merged**